go/database/mpt/proof_test.go

```diff
@@ -175,6 +175,101 @@ func TestCreateWitnessProof_CanCreateProof(t *testing.T) {
 	}
 }
 
+func TestCreateWitnessProof_ProofContainsEmptyNodes(t *testing.T) {
```

Comment:
I am not sure what this test is supposed to test. It does not check that proofs contain empty nodes (which it should not) nor does it check that it does not contain empty nodes.

Reply:
ok, I removed this test completely, when two out of three tests mimics an unrealistic MPT, it is not worth fixing it. Test of empty storage and empty trie is covered by other tests.

The other new test was extended to cover also a non-existing account - it is not passing at the moment, but will get fixed by the other PR - while the fix is obvious, I did not do it to prevent merge conflicts.